### ISSUE TITLE
Feature request: CommandPalette

### ISSUE THREAD

**Issue by zanmato** [longbridge/gpui-component#2748]
Command palettes are quite popular and are excellent for keyboard navigation, searching and quick actions that doesn't have a shortcut.

Based on https://ui.shadcn.com/docs/components/base/command

Most of the functionality already exists, notably `Dialog` and `searchable_list`. 

The app should own the shortcut,
```rust
cx.bind_keys([KeyBinding::new(
    "secondary-shift-p",
    ToggleCommandPalette,
    None,
)]);
```

The API could look something like this:

Basic usage, 
```rust
let palette = cx.new(|cx| {
    CommandPaletteState::new(
        CommandGroups::new([
            CommandGroup::new("Suggestions").with_items([
                CommandItem::new("New File", NewFile)
                    .with_icon(IconName::Plus)
                    .with_keywords(["create", "add"]),
                CommandItem::new("Save File", SaveFile),
            ]),
            CommandGroup::new("Account")
                .with_separator(true)
                .with_items([CommandItem::new("Sign Out", SignOut)]),
        ]),
        window,
        cx,
    )
});
```

With dynamic items,

```rust
CommandGroups::dynamic(|cx| {
    let mut groups = vec![CommandGroup::new("Suggestions").with_items([
        CommandItem::new("New Something", NewRequest),
    ])];

    for item in SomeManager::global(cx).read(cx).items() {
        groups.push(
            CommandGroup::new(item.name.clone())
                .with_separator(true)
                .with_items(item.sub_items.iter().map(|sub_item| {
                    CommandItem::new(
                        format!("{} {}", sub_item.method, sub_item.name),
                        DoSomething {
                            big: sub_item.name.clone(),
                        },
                    )
                    .with_id(format!("{}:{}", item.name, sub_item.name))
                })),
        );
    }

    groups
})
```
